sys-firmware/edk2/edk2-202502.ebuild

```diff
@@ -0,0 +1,345 @@
+		-D NETWORK_IP6_ENABLE \
+		-D NETWORK_TLS_ENABLE \
+		-D TPM1_ENABLE \
+		-D TPM2_ENABLE \
+		-D TPM2_CONFIG_ENABLE \
+		"${BUILD_ARGS[@]}" \
+		"${@}"
+}
+
+# Add the MS and Red Hat Secure Boot certificates and update the revocation list
+# for the given architecture in the given raw variables image.
+mk_fw_vars() {
+	edo virt-fw-vars \
+		--set-dbx "${DISTDIR}/$1_DBXUpdate_${DBXDATE}.bin" \
+		--secure-boot --enroll-redhat --inplace "$2"
+}
+
+# Convert the given images from raw to QCOW2 and resize them to the amount given
+# as the first argument. Specify 0 to not resize.
+raw_to_qcow2() {
+	local SIZE=$1 RAW
+	shift
+
+	for RAW in "${@}"; do
+		edo qemu-img convert -f raw -O qcow2 -o cluster_size=4096 -S 4096 "${RAW}" "${RAW%.fd}.qcow2"
+		[[ ${SIZE} != 0 ]] && edo qemu-img resize -f qcow2 "${RAW%.fd}.qcow2" "${SIZE}"
+		rm "${RAW}" || die
+	done
+}
+
+src_compile() {
+	TOOLCHAIN="GCC5"
+	BUILD_TARGET="RELEASE"
+	BUILD_DIR="${BUILD_TARGET}_${TOOLCHAIN}"
+	BUILD_ARGS=()
+
+	tc-export_build_env
+	emake -C BaseTools \
+		CC="$(tc-getBUILD_CC)" \
+		CXX="$(tc-getBUILD_CXX)" \
+		EXTRA_OPTFLAGS="${BUILD_CFLAGS}" \
+		EXTRA_LDFLAGS="${BUILD_LDFLAGS}"
+
+	export \
+		"${TOOLCHAIN}_${TARGET_ARCH}_PREFIX=${CHOST}-" \
+		"${TOOLCHAIN}_BIN=${CHOST}-"
+
+	. ./edksetup.sh
+
+	# DO NOT enable the shell with Secure Boot as it can be used as a bypass!
+
+	case "${ARCH}" in
+	amd64)
+		local SIZE
+		for SIZE in _2M _4M; do
+			mybuild -a X64 -p OvmfPkg/OvmfPkgX64.dsc \
+				-D FD_SIZE${SIZE}B \
+				-D BUILD_SHELL=FALSE \
+				-D SECURE_BOOT_ENABLE \
+				-D SMM_REQUIRE
+
+			mv -T Build/OvmfX64 Build/OvmfX64${SIZE}.secboot || die
+
+			mybuild -a X64 -p OvmfPkg/OvmfPkgX64.dsc \
+				-D FD_SIZE${SIZE}B
+
+			mv -T Build/OvmfX64 Build/OvmfX64${SIZE} || die
+
+			mk_fw_vars x64 Build/OvmfX64${SIZE}.secboot/"${BUILD_DIR}"/FV/OVMF_VARS.fd
+		done
+
+		# Fedora only converts newer images to QCOW2. 2MB images are raw.
+		raw_to_qcow2 0 Build/OvmfX64_4M*/"${BUILD_DIR}"/FV/OVMF_{CODE,VARS}.fd
+		;;
+	arm64)
+		BUILD_ARGS+=(
+			# grub.efi uses EfiLoaderData for code
+			--pcd PcdDxeNxMemoryProtectionPolicy=0xC000000000007FD1
+			# shim.efi has broken MemAttr code
+			--pcd PcdUninstallMemAttrProtocol=TRUE
+		)
+
+		mybuild -a AARCH64 -p ArmVirtPkg/ArmVirtQemu.dsc \
+			-D BUILD_SHELL=FALSE \
+			-D SECURE_BOOT_ENABLE
+
+		mv -T Build/ArmVirtQemu-AARCH64 Build/ArmVirtQemu-AARCH64.secboot_INSECURE || die
+
+		mybuild -a AARCH64 -p ArmVirtPkg/ArmVirtQemu.dsc
+
+		mk_fw_vars arm64 Build/ArmVirtQemu-AARCH64.secboot_INSECURE/"${BUILD_DIR}"/FV/QEMU_VARS.fd
+		raw_to_qcow2 64m Build/ArmVirtQemu-AARCH64*/"${BUILD_DIR}"/FV/QEMU_{EFI,VARS}.fd
+		;;
+	loong)
+		BUILD_ARGS+=(
+			# fails to seed the OpenSSL RNG during early initialization due
+			# to improper FPU enabling (maybe too late)
+			-D NETWORK_TLS_ENABLE=FALSE
+		)
+		mybuild -a LOONGARCH64 -p OvmfPkg/LoongArchVirt/LoongArchVirtQemu.dsc
+		raw_to_qcow2 16m Build/LoongArchVirtQemu/"${BUILD_DIR}"/FV/QEMU_{EFI,VARS}.fd
```

Comment:
You can put `0` here instead of `16m` since these files don't need to be padded.

Reply:
Done